vertexai/genai/chat.go

```diff
@@ -59,15 +59,28 @@ func (cs *ChatSession) SendMessageStream(ctx context.Context, parts ...Part) *Ge
 }
 
 // By default, use the first candidate for history. The user can modify that if they want.
-func (cs *ChatSession) addToHistory(cands []*Candidate) bool {
+func (cs *ChatSession) addToHistory(cands []*Candidate) {
 	if len(cands) > 0 {
 		c := cands[0].Content
 		if c == nil {
-			return false
+			return
 		}
-		c.Role = roleModel
-		cs.History = append(cs.History, c)
-		return true
+		cs.History = append(cs.History, copySanitizedModelContent(c))
```

Comment:
is it valid to append Content that could contain no parts? I am thinking if c.Parts only contains empty Text.

Reply:
Yes, I think it's OK for now, because this comes from the server. There should be an error somewhere if the server returns completely empty contents.